### PR TITLE
MaterialAutoComplete reworked to be more customizable and object oriented.

### PR DESCRIPTION
**Pull request by gilberto-torrezan**
Changed the way MaterialAutoComplete deals with the Suggestion object. Now it is the core of its functionality (instead of pure Strings), giving more control and flexibility to the user.

The MaterialAutoComplete now implements HasValue<List<? extends Suggestion>> as well, to be more GWT friendly.

### CODE REVIEW

**Comment by BenDol** [comment]
Great :+1: just need to replace the tabs with spaces (4 spaces).

File: gwt-material/src/main/java/gwt/material/design/client/ui/MaterialAutoComplete.java [hunk 1]
```diff
@@ -168,93 +219,99 @@ public void onSelection(SelectionEvent<SuggestOracle.Suggestion> selectionEvent)
...
-        }
+    	itemBox.setValue("");
+    	
+    	Collection<MaterialChip> values = suggestionMap.values();
+    	for (MaterialChip chip : values) {
+    		Widget parent = chip.getParent();
+    		if (parent instanceof ListItem){
+    			parent.removeFromParent();
+    		}
+		}
```
Comment (BenDol) [inline gwt-material/src/main/java/gwt/material/design/client/ui/MaterialAutoComplete.java]: Missing indent?

**Comment by BenDol** [comment]
Other than the few style things it LGTM.

**Comment by gilberto-torrezan** [comment]
Bloody hell, are you serious you use 4 spaces instead of tab?

http://stackoverflow.com/research/developer-survey-2015#tech-tabsspaces

Anyway, if code style is a show stopper to contributing, you should make your policies clear somewhere on the site =/

**Comment by gilberto-torrezan** [comment]
Ok, now I'm using the GWT code style conventions, and added a note on README about that. If you decide to use any other convention, or to create your own, please notify me so I can reformat the code.

The full project is not using GWT code style (just the classes I pushed), so if you want to stick with that, a full reformat will be needed.

**Comment by kevzlou7979** [comment]
@gilberto-torrezan  Thanks for the Autocomplete revision to implement OOP concept inside the component.

**Comment by remarkNS** [comment]
I think we should open new ticket for discussion about this conventions. Some people are using Eclipse (like me), some of them Idea or Netbeans. Some people like to use tabs (like me) some of them like 4 spaces instead. I think it does not matter which convention is used as long as majority of people agrees with it in a friendly discussion. What do you think guys ?

**Comment by kevzlou7979** [comment]
Yes I agree we need to discuss this coding convention for better coding style strategy

**Pull request merged**